tests/test_fmi_master.py

```diff
@@ -394,6 +395,137 @@ def do_step2(current_t, step_size, new_step=True):
+        downsample_indices[0] = True
+        downsample_indices[-1] = True
+
+        np.testing.assert_equal(ref_trajs[0][downsample_indices], test_trajs[0])
+        np.testing.assert_equal(ref_trajs[1][downsample_indices], test_trajs[1])
+
+    def test_downsample_result(self):
+        """ Test multiple result_downsampling_factor value and verify the result. """
+        opts_update = {'result_downsampling_factor': 1}
+        models, connections = self._load_basic_simulation()
+        ref_res = self._sim_basic_simulation(models, connections, opts_update)
+
+        ref_traj_length = 2001
+        assert len(ref_res[models[0]]['time']) == ref_traj_length
+        assert len(ref_res[models[1]]['time']) == ref_traj_length
+
+        ref_res_trajs = (ref_res[models[0]]['x1'].copy(), ref_res[models[1]]['x2'].copy())
+
+        for f in [2, 3, 4, 5, 10, 100, 250, 600, 3000]:
+            opts_update = {'result_downsampling_factor': f}
+            models, connections = self._load_basic_simulation()
+            test_res = self._sim_basic_simulation(models, connections, opts_update)
+
+            test_res_trajs = (test_res[models[0]]['x1'], test_res[models[1]]['x2'])
+            self._verify_downsample_result(ref_res_trajs, test_res_trajs, 2000, f)
+
+    def test_downsample_error_check_invalid_value(self):
+        """ Verify we get an exception if the option is set to anything less than 1. """
+        models, connections = self._load_basic_simulation()
+        test_values = [-10, -20, -1, 0]
+
+        # TODO: tidy up with pytest
+        expected_substr = "Valid values for option 'result_downsampling_factor' are only positive integers"
+        for value in test_values:
+            try:
+                self._sim_basic_simulation(models, connections, {'result_downsampling_factor': value})
+                error_raised = False
+            except FMUException as e:
+                error_raised = True
+                assert expected_substr in str(e), f"Error was {str(e)}, expected substring {expected_substr}"
+            assert error_raised
+
+    def test_error_check_invalid_value(self):
+        """ Verify we get an exception if the option is set to anything that is not an integer. """
+        models, connections = self._load_basic_simulation()
+        test_values = [1/2, 1/3, "0.5", False]
+
+        # TODO: tidy up with pytest
+        expected_substr = "Option 'result_downsampling_factor' must be an integer,"
+        for value in test_values:
+            try:
+                self._sim_basic_simulation(models, connections, {'result_downsampling_factor': value})
+                error_raised = False
+            except FMUException as e:
+                error_raised = True
+                assert expected_substr in str(e), f"Error was {str(e)}, expected substring {expected_substr}"
+            assert error_raised
+        
+    # TODO: Test case that supports storing FMU states required
```

Comment:
I don't think we have any example that would work here readily available and I am not sure if it is worth it to construct one here just to test the warning.

Reply:
I agree